### ISSUE TITLE
More dev docs please :)

### ISSUE THREAD

**Issue by sidneymbell** [nextstrain/augur#450]
Basically preferred communication channels and processes. 

E.g., if you're an outside dev, can you just self-assign an issue? who should you check with first? do you have a fixed release cycle? should you get approval from the core team for your implementation plan before starting to code? etc.

**Comment by tsibley** [comment]
By putting these in the conventional `CONTRIBUTING.md` file for each repo, GitHub will surface them in its UI.

**Comment by tsibley** [comment]
Sidney, could you maybe draft up such a file? :grin: My quick thoughts:

- Self-assigning an issue or just commenting that you're working on it is :+1:.
- Good to check in first by commenting on an issue, asking questions there, and sharing implementation plans. If there's not an existing issue, making one for a bug you see is definitely ok.
- No fixed release cycle. We do care about backwards compat and use semver.

Good to get quick thoughts of others too, like @emmahodcroft @rneher @huddlej @jameshadfield. I think @kairstenfay also was doing a contributing doc for Auspice?

**Comment by emmahodcroft** [comment]
Can we brainstorm maybe some guidance on the nextstrain way to do things..? To me this may sometimes be hardest.. That our audience and grandad userbase is generally scientists, not all who have a strong programming background, that we care a lot about representing uncertainty because this is really important in disease.stuff... To get people into the right mindset? I hope that makes sense.
Tom maybe you are particularly well placed for any comments, I know sometimes we have pushed back against ideas that seem very sensible from a coding perspective for odd reasons 🙃 though we try to work with! But that's the kind of area where it might be nice to have people prepped... As I imagine that might be most time consuming to explain over and over

**Comment by jameshadfield** [comment]
For the Auspice, Augur & Nextstrain.org repos we should collect issues which are well defined problems, ones which have a clear path to fixing, and tag these appropriately.

**Comment by sidneymbell** [comment]
Awesome, thanks for your input everyone! I'd propose the following:  

- [ ] I'll draft up some contributor guidelines and submit a PR for @tsibley's review  

- [ ] I'll also take a first stab at [a values statement](https://docs.google.com/presentation/d/1micFSwi3ywML7ck3clQHAUf2LzBaTUlW49CzAZTXGio/edit?usp=sharing) that captures the sorts of (super important) things @emmahodcroft described. We can then all iterate on this until we feel comfortable cross-posting it to the relevant repositories.

- [ ] @jameshadfield -- I completely agree that having issues tagged as "good first issue" and/or "ready for implementation" would be a HUGE benefit for welcoming new devs into the community in a comfortable way. Can you please make an issue for this and assign the relevant folks? (I know a lot of motivated engineers with some extra time on their hands -- having an "architecture of participation" can potentially be a big benefit)

**Comment by kairstenfay** [comment]
>  I think @kairstenfay also was doing a contributing doc for Auspice?

Yes, I started a [CONTRIBUTING.md](https://github.com/nextstrain/auspice/blob/master/CONTRIBUTING.md) for Auspice. I

**Comment by kairstenfay** [comment]
I think a flow chart or high-levlel technical description of how Auspice works would be helpful for new contributors, too. @jameshadfield gave such a talk to Jover and I when he visited Seattle.

**Comment by emmahodcroft** [comment]
@jameshadfield  also did a lot of work last night cleaning up issues and maknig them more specific!

**Comment by sidneymbell** [comment]
Awesome!!!

**Comment by sidneymbell** [comment]
@kairstenfay -- this is awesome. Is it helpful for me to start on one for augur, or would you prefer to?

**Comment by jameshadfield** [comment]
Last night I updated most of the readmes, added a sentance linking to https://nextstrain.org/docs/contributing/development to the nextstrain front page, and all of the above link to https://github.com/orgs/nextstrain/projects/5 which i'll keep up-to-date (bear in mind time is precious right now so all help in maintaining and communicating is most welcome!).

**Comment by jameshadfield** [comment]
> I think a flow chart or high-level technical description of how Auspice works would be helpful for new contributors, too. @jameshadfield gave such a talk to Jover and I when he visited Seattle.

@kairstenfay @joverlee521 can you make this into a document somewhere do you think? Agree that the bar to starting with auspice is high.

**Comment by kairstenfay** [comment]
> @kairstenfay @joverlee521 can you make this into a document somewhere do you think? Agree that the bar to starting with auspice is high.

While necessary, I don't think Jover or I will have time anytime soon to work on this. Things directly related to COVID-19 will be taking up our time for the foreseeable future.

**Comment by huddlej** [comment]
@sidneymbell Is the values statement similar to a "code of conduct"? I've been trying to learn more about this side of OSS by reading [NumFOCUS's project sponsorship requirements](https://numfocus.org/projects-overview) which includes a code of conduct. The pandas project has a [simple document to fulfill this requirement](https://github.com/pandas-dev/pandas-governance/blob/master/code-of-conduct.md). Would something like this live next to a values statement or in the same file?

Also, there was discussion in [this related nextstrain.org issue](https://github.com/nextstrain/nextstrain.org/issues/128) of having a Nextstrain-wide CONTRIBUTING guide. Would you want to use your draft for this issue as a starting point for that document? Is there any way I can help you with writing/reviewing this?

**Comment by sidneymbell** [comment]
Closing as it's a duplicate of [this more comprehensive issue](https://github.com/nextstrain/nextstrain.org/issues/128)
